### DISCUSSION TITLE
Customisable Sidebar

### DISCUSSION THREAD

**Discussion by onnyyonn** [osmoscraft/osmosfeed#36] [Ideas]
A customisable sidebar would be really handy for adding useful features such as site-wide search, navigating to a particular day/month, filtering option (like show "fresh" articles only), keyboard shortcut lists etc. Bonus if the sidebar is theme-able with the template API and users can choose which feature-module to add. Some extra modifications are needed for smaller screen sizes, where it's probably better to keep the sidebar hidden by default.

**Comment by chuanqisun** [comment]
Love the idea. I'll do some exploration. This can be a great solution to #35. I think the current challenge is to design a stable interface so the sidebar widget can function across different templates. Since HTML is the only shared interface between templates, I'll need to set a standard on how the template should render the item so they can be programmatically controlled by widgets.

For example, whenever we render an article, require a machine readable date on the article level:
```hbs
<article data-iso-date="{{isoDate}}">...</article>
```
